### PR TITLE
Makefile: add install rule

### PR DESCRIPTION
**Pull request by jowong4**
Add install rule to Makefile

### CODE REVIEW

**Comment by lcoombe** [comment]
Just curious -- the binaries will be installed to that specified location, but I guess the users will likely never use them there because the Makefile specifically points to the `src` directory in the repo?

**Comment by jowong4** [comment]
Yeah, that is currently the case. I'm guessing this is intended for the more advanced users who want to use/configure parts of physlr so they add the install path to their `PATH` variable. Can call the binaries individually.

File: src/Makefile [hunk 1]
```diff
@@ -1,5 +1,8 @@
 CXXFLAGS=-g -O3 -Wall -Werror -Wno-error=unknown-pragmas -std=c++11 -pthread -I./include -I./vendor -fopenmp
 LDLIBS=-lz
+PREFIX = $(HOME)/.local
```
Comment (sjackman) [inline src/Makefile]:
```suggestion
PREFIX=$(HOME)/.local
```

File: src/Makefile [hunk 2]
```diff
@@ -65,3 +68,7 @@ check-physlr-makebf: all
 check-physlr-overlap: all
 	./physlr-overlap -t4 -n1 data/tiny.overlap.input.tsv | sort | diff -q - data/tiny.physlr.overlap.n1.sorted.tsv
 	./physlr-overlap -t4 -n2 data/tiny.overlap.input.tsv | sort | diff -q - data/tiny.physlr.overlap.n2.sorted.tsv
+
+install: physlr-indexlr physlr-filter-barcodes physlr-overlap physlr-filter-bxmx physlr-makebf
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
```
Comment (sjackman) [inline src/Makefile]:
```suggestion
	install -d $(DESTDIR)$(PREFIX)/bin
```

File: src/Makefile [hunk 3]
```diff
@@ -65,3 +68,7 @@ check-physlr-makebf: all
...
+	for f in $^; do cp $$f $(DESTDIR)$(PREFIX)/bin; done
```
Comment (sjackman) [inline src/Makefile]:
```suggestion
	install $^ $(DESTDIR)$(PREFIX)/bin
```

File: src/Makefile (hunk 2 above)
Comment (jowong4) [reply src/Makefile]: i used `install $^ -t $(DESTDIR)$(PREFIX)/bin` instead. Is that alright?

File: src/Makefile (hunk 3 above)
Comment (jowong4) [reply src/Makefile]: see above

File: src/Makefile (hunk 2 above)
Comment (jowong4) [reply src/Makefile]: Nevermind, the above command doesn't seem to make the directory if it doesn't exist

**Pull request merged**